packages/mongodb-rag-core/src/dataSources/normalizeUrl.ts

```diff
@@ -9,3 +9,14 @@ export function normalizeUrl(url: string): string {
     .replace(/^www\./, "")
     .replace(/\/+$/, "");
 }
+
+/**
+  Adds protocol ("https://") to a URL if it is missing. 
+  Intended for use in URL constructor.
+ */
+export function ensureProtocol(url: string): string {
+  if (url.startsWith("http://") || url.startsWith("https://")) {
+    return url;
+  }
+  return `https://${url}`;
+}
```

Comment:
can u see if there's a way to do this with the `URL` module? would be more resilient to edge cases, like malformatted urls. think this API should work https://developer.mozilla.org/en-US/docs/Web/API/URL/protocol